### DISCUSSION TITLE
adding webapp in scene

### DISCUSSION THREAD

**Discussion by lalitupmanyu** [mozilla/hubs#5043] [Spoke Questions] [answered]
I want to add a webapp in room e.g. https://psim.shinyapps.io/business_game/

when added as link, it does not run in room but opens a new tab in browser.

Is there a way, this can be run within room like it is done for videos/youtube links.

Thanks
Lalit

**Comment by MozillaHubsFriend** [comment, marked as answer]
A simple workaround would be to use screenshare and share the tab corresponding to the web app.

If you wanted to interact with it in-room, you would need to make edits to the code with hubs cloud. There is a branch here that adds iframes to hubs: https://github.com/mozilla/hubs/tree/feature/iframes but there are some limitations with iframes.

**Reply by postelontwerpen** [reply]
I m also really interested in a solution for loading clickable webpages/pwa’s. Because of missing interaction options in hubs itself, building a webapp seems a flexible way to give users much more interactive experience, like small games, ordering products, do a quiz and more. Now an url is always a canvas in Hubs, so not clickable, only a link to the website itself. But maybe it’s such thing not possible in a browser or a big risk for big pageload in Hubs?
